frontend/public/components/silence.tsx

```diff
@@ -0,0 +1,200 @@
+              <dl className="co-m-pane__details">
+                {silence.name && <React.Fragment>
+                  <dt>Name</dt>
+                  <dd>{silence.name}</dd>
+                </React.Fragment>}
+                <dt>State</dt>
+                <dd><State silence={silence} /></dd>
+                <dt>Last Updated At</dt>
+                <dd><Timestamp timestamp={silence.updatedAt} /></dd>
+              </dl>
+            </div>
+            <div className="col-sm-6">
+              <dl className="co-m-pane__details">
+                <dt>Starts At</dt>
+                <dd><Timestamp timestamp={silence.startsAt} /></dd>
+                <dt>Ends At</dt>
+                <dd><Timestamp timestamp={silence.endsAt} /></dd>
+                <dt>Created By</dt>
+                <dd>{silence.createdBy || '-'}</dd>
+                <dt>Comments</dt>
+                <dd>{silence.comment || '-'}</dd>
+                <dt>Silenced Alerts</dt>
+                <dd>{numSilencedAlerts === undefined ? '-' : numSilencedAlerts}</dd>
+              </dl>
+            </div>
+          </div>
+        </div>
+      </div>
+    </StatusBox>
+  </React.Fragment>;
+});
+export const SilencesDetailsPage = connectMonitoringPage(SilencesDetailsPage_);
+
+const rowStateToProps = (state, {obj}) => {
+  const {data}: Alerts = monitoringAlertsToProps(state);
+  const labels = silenceLabels(obj);
+  return {numSilencedAlerts: data ? _.filter(data, _.matches({labels})).length : undefined};
+};
+
+const SilenceRow = connect(rowStateToProps)((props: SilenceRowProps) => {
+  const {numSilencedAlerts, obj} = props;
+  const state = silenceState(obj);
+
+  return <ResourceRow obj={obj}>
+    <div className="col-xs-7">
+      <div className="co-resource-link-wrapper">
+        <div className="co-resource-link">
+          <MonitoringResourceIcon resource={SilenceResource} />
+          <Link className="co-resource-link__resource-name" title={obj.id} to={`${SilenceResource.path}/${obj.id}`}>{obj.name}</Link>
+        </div>
+      </div>
+    </div>
+    <div className="col-xs-3">
+      <State silence={obj} />
+      <div className="text-muted monitoring-timestamp">
+        {state === 'pending' && <React.Fragment>starts&nbsp;<Timestamp timestamp={obj.startsAt} /></React.Fragment>}
+        {state === 'active' && <React.Fragment>ends&nbsp;<Timestamp timestamp={obj.endsAt} /></React.Fragment>}
+        {state === 'expired' && <React.Fragment>expired&nbsp;<Timestamp timestamp={obj.endsAt} /></React.Fragment>}
+      </div>
+    </div>
+    <div className="col-xs-2">{numSilencedAlerts === undefined ? '-' : numSilencedAlerts}</div>
+  </ResourceRow>;
+});
+
+const SilenceHeader = props => <ListHeader>
+  <ColHead {...props} className="col-xs-7" sortField="name">Name</ColHead>
+  <ColHead {...props} className="col-xs-3" sortField="status.state">State</ColHead>
+  <ColHead {...props} className="col-xs-2" sortField="labels.severity">Silenced Alerts</ColHead>
```

Comment:
This sort field doesn't look right

Reply:
Thanks. Actually, this sort is a bit tricky to implement, so I just removed the sort for now.